### DISCUSSION TITLE
FR: Include title, url_title, Structure fields in conditional fields conditions

### DISCUSSION THREAD

**Discussion by tyssen** [ExpressionEngine/ExpressionEngine#2929] [Feature Requests]
I'm working on a site that uses Structure and a lot of the site's pages are built with a single `content` channel using a page builder field. Different templates can be assigned with Structure's settings.

Sometimes, a need comes up for just one page to have a single different field from all the rest. At the moment, the options for implementing this are either:

- Add the field to the `content` channel which means that it appears on every page's publish layout, even though it's only actually used for one page.
- Create a new channel and add the new field to it just for the one entry.
- Create a separate settings channel that has just one entry in it to which you add all of these little ad-hoc settings. So the editor needs to make changes in two different places.
- Same thing as above but use Low Variables instead (you still need to edit in two different places).

When discussing one such change yesterday I wondered whether we could use conditional fields instead. It turns out we can't because the only fields that have data in them that differentiate the page we want to add the conditional field to are `title`, `url_title`, `page_uri`, etc. but they're not available. If they were, conditional fields would be the answer to this problem.

**Comment by intoeetive** [comment]
Copying over similar request by @melbazany from https://github.com/ExpressionEngine/ExpressionEngine/issues/2750

> Would love the ability to conditionally display a field based on categories selected:
> 
> Ex: If any category is selected, show X field.
> Ex: If category named 'Foo Bar' is selected, show X field.
> Ex: If at least 1 category from group X is selected, show X field.
> 
> At this point, I've just been adding a Notes field to the channel entries to make sure users are aware of the condition in order for a field to be utilized properly..

**Comment by intoeetive** [comment]
Requested by @shbchk in https://github.com/ExpressionEngine/ExpressionEngine/discussions/3249
> Ability to use Entry Status in Conditional Fields

**Comment by intoeetive** [comment]
Field Conditions are currently field_id - based, as those are stored in DB table that's using field_id as foreign key.

The possible solution would be to add models like `CategoryCondition`, `TitleCondition` etc and add the code to process those.

**Comment by intoeetive** [comment]
This might be not finalized yet, but if anyone wants to give a try: https://github.com/ExpressionEngine/ExpressionEngine/pull/3854
